src/bun.js/api/server.zig

```diff
@@ -5796,6 +5796,177 @@ pub const ServerWebSocket = struct {
+                        return;
+                    },
+                    .fulfilled => {
+                        handleOnResolve(this);
+                        return;
+                    },
+                    .rejected => {
+                        // const value = promise.asValue(globalThis);
+                        const value = promise.result(globalThis.vm());
+                        handleOnReject(this, globalThis, value);
+                        return;
+                    },
+                }
+            }
+
+            if (result.toError()) |e| {
+                handleOnReject(this, globalThis, e);
+            } else {
+                handleOnResolve(this);
+            }
+        }
+    }
+
+    pub fn deinit(this: *ServePlugins) void {
+        if (this.value == .err) {
+            bun.default_allocator.free(this.value.err);
+        }
+        ServePlugins.destroy(this);
+    }
+
+    pub const onResolve = JSC.toJSHostFunction(onResolveImpl);
+    pub const onReject = JSC.toJSHostFunction(onRejectImpl);
+
+    pub fn onResolveImpl(_: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) bun.JSError!JSValue {
+        ctxLog("onResolve", .{});
+
+        const arguments = callframe.arguments_old(2);
+        var plugins = arguments.ptr[1].asPromisePtr(ServePlugins);
+        defer plugins.deref();
+        const plugins_result = arguments.ptr[0];
+        plugins_result.ensureStillAlive();
+
+        handleOnResolve(plugins);
+
+        return JSValue.jsUndefined();
+    }
+
+    pub fn handleOnResolve(this: *ServePlugins) void {
+        this.value.pending.promise.deinit();
+        for (this.value.pending.pending_bundled_routes.items) |route| {
+            route.onPluginsResolved(this.value.pending.plugins.?);
+            route.deref();
+        }
+        this.value.pending.pending_bundled_routes.clearRetainingCapacity();
+        this.value = .{ .result = this.value.pending.plugins.? };
+    }
+
+    pub fn onRejectImpl(globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) bun.JSError!JSValue {
+        ctxLog("onReject", .{});
+
+        const arguments = callframe.arguments_old(2);
+        const plugins = arguments.ptr[1].asPromisePtr(ServePlugins);
+        handleOnReject(plugins, globalThis, arguments.ptr[0]);
+
+        return JSValue.jsUndefined();
+    }
+
+    pub fn handleOnReject(plugins: *ServePlugins, globalThis: *JSC.JSGlobalObject, e: JSValue) void {
+        defer plugins.deref();
+        const bunstr = e.toBunString(globalThis);
+        defer bunstr.deref();
+        const str = bunstr.toUTF8(bun.default_allocator);
+        defer str.deinit();
+        const str_slice = str.slice();
+        for (plugins.value.pending.pending_bundled_routes.items) |route| {
+            route.onPluginsRejected(str_slice);
+            route.deref();
+        }
+        plugins.value.pending.pending_bundled_routes.clearRetainingCapacity();
```

Comment:
Ideally we would let the user be able to intercept this and handle it via the `onError` callback in Bun.serve().

IF we cannot do that for timing reasons for now, then we can instead use vm.onUnhandledRejection and pass the JSValue to that. 

We should not be stringifying